lib/Twig/Extensions/Extension/I18n.php

```diff
@@ -10,6 +10,18 @@
  */
 class Twig_Extensions_Extension_I18n extends Twig_Extension
 {
+    /** @var Twig_Extensions_Translator */
+    private $translator;
+
+    public function __construct(Twig_Extensions_Translator $translator = null)
+    {
+        if ($translator === null) {
+            $this->translator = new Twig_Extensions_NativeGettextTranslator();
```

Comment:
I know it's a bad practice to call a constructor within a constructor, I implemented it this way to be backwards compatible. Perhaps in future versions the null default value for ´$translator´ can be dropped together with the first branch of this if block. Then one would have to initialize the Extension like this for the native gettext implementation:
```php
$twig->addExtension(new Twig_Extensions_Extension_I18n(new Twig_Extensions_NativeGettextTranslator()));
```